src/Requests.jl

```diff
@@ -153,6 +156,11 @@
         r.finished = true
         close(p.sock)
 
+        #If MIME type allows, convert r.data to a Julia string
+        mimetype = r.headers["Content-Type"]
+        if (contains(mimetype, "charset=utf-8") || contains(mimetype, "charset=us-ascii"))
```

Comment:
I see a problem in an inconsistency of a returned body type. If I would not be able predict type of a returned structure, I'll be forced to write additional safeguards. In this case I would prefer to have a binary output which I'll be able convert to `string` based on a program logic or additional information from a parsed header rather then check type of the body and reconvert its type. For me, it is a matter of a data flow control.